confgenerator/confgenerator_test.go

```diff
@@ -49,13 +49,20 @@ var (
 )
 
 var platform string
+var hostInfo *host.InfoStat
 
 func init() {
-	hostInfo, _ := host.Info()
+	hostInfo, _ = host.Info()
 	if hostInfo.OS == "windows" {
```

Comment:
The test shouldn't be relying on the OS - just hardcode one string and use that in the tests regardless of which platform you're on.

Reply:
It has to be relying on OS. See this PR: https://github.com/GoogleCloudPlatform/ops-agent/pull/49
Reasons are also listed below
- file path is different on Linux v.s. Windows
- collectd sends its own logs to a file, thus fluent-bit tails that file which  is unique on Linux v.s. windows otel will send its own logs to eventlog
- the ending \r\n is windows v.s. \n is linux

Reply:
I understand the output is different, but the OS should be an input to the confgenerator and the tests should hardcode one (or both) choice for OS.

Reply:
Now travis runs this confgenerator.test on two different platforms(Linux and Windows): https://github.com/GoogleCloudPlatform/ops-agent/blob/master/.travis.yml. And it's using the real hostinfo of the running platform

Reply:
I've been struggling a bit with triggering Windows unit tests from a Linux dev machine (it's a bit painful to always have to start a Windows VM to run Windows unit tests). Quentin's proposal is similar to what I have in mind to improve that experience, and I was planning to send out a PR. It's probably better to address this outside this current PR though as it's not directly related, right?

Reply:
Okay then, we shall address it in a separate PR once we decide on how we want to do the test *across platform*